corehq/form_processor/submission_post.py

```diff
@@ -328,7 +320,12 @@ def run(self):
                 elif not instance.is_error:
                     submission_type = 'normal'
                     try:
-                        case_stock_result = self.process_xforms_for_cases(xforms, case_db, self.timing_context)
+                        case_stock_result = self.process_xforms_for_cases(xforms, case_db,
+                                                                          self.domain, self.timing_context)
+                    except MetadataError as e:
+                        instance.problem = e
+                        instance.state = XFormInstance.ERROR
+                        submission_type = 'error'
```

Comment:
This should be incorporated into the 'known errors' block below which will give you the standard error handling (you are currently not saving the form and not responding with the correct OpenRosa error nature).

Reply:
I'm reluctant to use `_handle_known_error` since the whole point of this validation is to prevent forms with missing tags/nodes from being saved in the first place, which is what seems to be doing, regardless of what the error is. Though I may be misunderstanding what it does since I assume we also don't want forms with other exceptions (IllegalCaseId, UsesReferrals, MissingProductId, etc.) to be saved, but we do anyways by calling _handle_known_error??